File: memory-domain/src/main/java/com/memory/domain/game/repository/GameQuestionRepositoryCustom.java
package com.memory.domain.game.repository;

import com.memory.domain.game.GameQuestion;
import com.memory.domain.game.GameSession;

import java.util.List;
import java.util.Optional;

public interface GameQuestionRepositoryCustom {

    Integer findNextQuestionOrder(GameSession gameSession);

    List<GameQuestion> findByGameSessionOrderByQuestionOrder(GameSession gameSession);

    Optional<GameQuestion> findByIdAndGameSession(Long questionId, GameSession gameSession);

}

File: memory-domain/src/main/kotlin/com/memory/domain/game/repository/GameQuestionRepositoryCustom.kt
package com.memory.domain.game.repository

import com.memory.domain.game.GameQuestion
import com.memory.domain.game.GameSession
import java.util.Optional

interface GameQuestionRepositoryCustom {

    fun findNextQuestionOrder(gameSession: GameSession?): Int?

    fun findByGameSessionOrderByQuestionOrder(gameSession: GameSession?): List<GameQuestion>

    fun findByIdAndGameSession(questionId: Long?, gameSession: GameSession?): Optional<GameQuestion>

}
